packages/contracts/tasks/helpers/ProofGenerator.ts

```diff
@@ -432,4 +511,101 @@ export class ProofGenerator {
 
     return proofs;
   }
+
+  /**
+   * Generate a proof for processing messages
+   * @param pollId - The ID of the poll
+   * @param incremental - Whether to use incremental proof generation
+   * @returns The proof for processing messages
+   */
+  async generateProcessMessagesProof(pollId: bigint, incremental = false): Promise<Proof> {
+    const poll = this.poll as unknown as MaciState;
+    const foundPoll = poll.polls.get(pollId);
+    if (!foundPoll) {
+      throw new Error(`Poll ${pollId} not found`);
+    }
+
+    // Load state if incremental mode is enabled
+    if (incremental) {
+      const statePath = path.join(this.outputDir, `poll_${pollId}_state.json`);
+      if (fs.existsSync(statePath)) {
+        await foundPoll.loadState(statePath);
+      }
+    }
+
+    // Process messages
+    foundPoll.processMessages(pollId, this.useQuadraticVoting, incremental);
+
+    // Save state if incremental mode is enabled
+    if (incremental) {
+      const statePath = path.join(this.outputDir, `poll_${pollId}_state.json`);
+      await foundPoll.saveState(statePath);
+    }
+
+    // Generate proof
+    const proofPath = path.join(this.outputDir, `processMessages_${pollId}.json`);
+    if (incremental && fs.existsSync(proofPath)) {
+      const existingProof = JSON.parse(await fs.promises.readFile(proofPath, "utf8")) as Proof;
+      return existingProof;
+    }
+
+    const circuitInputs = foundPoll.processMessages(pollId, this.useQuadraticVoting, incremental);
+    const proofs = await this.generateProofs(
+      circuitInputs as unknown as CircuitInputs, 
+      this.mp, 
+      `processMessages_${pollId}.json`, 
+      await extractVk(this.mp.zkey, false)
+    );
+    const proof = proofs[0];
+    await fs.promises.writeFile(proofPath, JSON.stringify(proof, null, 2));
+    return proof;
+  }
+
+  /**
+   * Generate a proof for tallying votes
+   * @param pollId - The ID of the poll
+   * @param incremental - Whether to use incremental proof generation
+   * @returns The proof for tallying votes
+   */
+  async generateTallyProof(pollId: bigint, incremental = false): Promise<Proof> {
```

Comment:
The logic for tally proof is the same as for message processing, it makese sense to reuse it and do not duplicate.